docs/distribution/launcher.md

```diff
@@ -0,0 +1,208 @@
+
+## Launcher Build
+The launcher is built using
+[GraalVM Native Image](https://www.graalvm.org/docs/reference-manual/native-image/)
+which compiles the JVM code into a native binary ahead of time, resulting in a
+small and fast launching executable.
+
+### Portability
+On Linux, it is possible to statically link all libraries required by the Native
+Image, thus ensuring portability between Linux distributions.
+
+On Windows and Mac, it is not possible to statically link against system
+libraries, but this should not hinder portability as the system libraries are
+generally compatible between distribution versions on these platforms.
+Non-system dependencies are included in the binary on these platforms as well.
+
+## Project Management
+The launcher provides basic project management utilities for the command-line
+user.
+
+### Creating a Project
+It allows to create an empty project in a specified directory with the basic
+configuration based on user's config.
+
+### Per-Project Enso Version
+Project configuration can specify the exact Enso version that should be used
+inside that project. The launcher automatically detects if it is in a project
+(by traversing the directory structure). The current project can also be
+specified by the `--path` parameter. All components launched inside a project
+use the version specified in the project configuration, or outside a project,
+the default version.
+
+> The actionables for this section are:
+> 
+> - Decide how to support inexact project bounds (like `>=3.1, <4` - when should
+>   the launcher check for new versions) and resolvers.
+> - Decide how to support nightly builds.
+
+### Project Configuration
+The command-line allows to edit project configuration, for example: change the
+author's name or Enso version.
+
+## Enso and Graal Version Management
+The launcher automatically manages required Enso versions. When running inside a
+project, if the version specified in project configuration is not installed, it
+is installed automatically.
+
+If an Enso version is no longer needed, it can be removed with the `uninstall`
+command.
+
+Moreover, GraalVM distributions tied to the installed Enso versions are managed
+automatically by the launcher. When a new Enso version is installed, it also
+ensures that the correct GraalVM version is installed and that it is used for
+launching this version of Enso. When a managed GraalVM distribution is no longer
+used by any installed version of Enso, it is automatically removed.
+
+### GraalVM Override
+While the launcher manages its own installation of GraalVM to ensure that the
+right JVM version is used to launch each version of Enso, the user can override
+this mechanism to use the installed system JVM instead. This is an advanced
+feature and should rarely be used.
+
+The launcher will check the system JVM and refuse to launch Enso if it is not a
+GraalVM distribution. It will also print a warning if the major version is
```

Comment:
It should also print a warning when minor version is different. Graal is not semvered and the API keeps changing between minor versions.